docs/data-migration.md

```diff
@@ -0,0 +1,238 @@
+# Data migration
+
+This document describes the process of migrating `LND`'s database state from one
+type of database backend (for example the `bbolt` based database files `*.db`
+such as the `channel.db` or `wallet.db` files) to another (for example the new `postgres` or `sqlite` databases introduced in `lnd v0.14.0-beta`).
+
+**Note:** Currently only migrations from `bolt` to either `postgres` or `sqlite` are supported. KV based databases will be phased out in the future. We are planning to add additonal support to migrate all `etcd` databases to `sql` databases. Moreover it is currently also not supported to move the database from `sqlite` to `postgres` or vice versa, because this migration only takes care of the key value data and there already exist the possiblity to have the invoices in native sql which this tool does not support.
+
+
+## Prepare the destination database
+
+### Using postgres as the destination remote database
+
+Prepare a user and database as described in the [Postgres](
+   https://github.com/lightningnetwork/lnd/blob/master/docs/postgres.md)
+documentation. You'll need the Data Source Name (DSN) for both the data
+migration and then the `lnd` configuration, so keep that string somewhere
+(should be something with the format of `postgres://xx:yy@localhost:5432/zz`).
+
+No additional steps are required to prepare the Postgres database for the data
+migration. The migration tool will create the database schema automatically, so
+no DDL scripts need to be run in advance.
+
+### Using sqlite as the destination remote database
+
+No particular preparation is needed for `sqlite` compared to the `postgres`
+case. Similar to the `bolt` case there will be separated db files created for
+each individual `bolt` database.
+
+
+
+## Prepare the source database
+
+Assuming we want to migrate the database state from the pre-0.19.0 individual
+`bbolt` based `*.db` files to a remote database, we first need to make sure the
+source files are in the correct state.
+
+The following steps should be performed *before* running the data migration (
+some of them are marked as optional and can be neglected for small databases
+e.g. 200MB
+):
+1. Stop `lnd`
+2. Upgrade the `lnd` binary to the latest version (e.g. `v0.19.0-beta` or later)
+3. (optional) Make sure to add config options like 
+   `gc-canceled-invoices-on-startup=true` and `db.bolt.auto-compact=true` to 
+   your `lnd.conf` to optimize the source database size by removing canceled 
+   invoices and compacting it on startup.
+4. Also make sure to migrate the revocation log for all channels active prior
```

Comment:
is there a way to enforce the flag? I think once the migration is done, `--db.prune-revocation` won't work as it's tailored for kvdb, so better make sure the data is already pruned before the migration.

Reply:
Yeah it will still work after the migration but we can force it here for users to speed up the migration, I am going to force it.

Reply:
I think almost all users should be on the new revocation log schema anyways.

Reply:
There is really no easy way to tell if the user actually has to set this config setting because channels might all be new (after `lnd@v0.15.0`) so I will remove this check and keep it optional.